File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Two_Pointers/Valid_Palindrome/valid_palindrome.ts
function isPalindrome(s: string): boolean {
    let left = 0;
    let right = s.length - 1;
    
    while (left < right) {
        while (left < right && !isAlphanumeric(s[left])) {
            left++;
        }
        
        while (left < right && !isAlphanumeric(s[right])) {
            right--;
        }
        
        if (s[left].toLowerCase() !== s[right].toLowerCase()) {
            return false;
        }
        
        left++;
        right--;
    }
    
    return true;
};

function isAlphanumeric(char: string): boolean {
    const code = char.charCodeAt(0);
    return (code >= 48 && code <= 57) ||
           (code >= 65 && code <= 90) ||
           (code >= 97 && code <= 122);
}

File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Two_Pointers/Valid_Palindrome/valid_palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int left = 0;
        int right = s.Length - 1;
        
        while (left < right) {
            // Skip non-alphanumeric from left
            while (left < right && !char.IsLetterOrDigit(s[left])) {
                left++;
            }
            
            // Skip non-alphanumeric from right
            while (left < right && !char.IsLetterOrDigit(s[right])) {
                right--;
            }
            
            // Compare case-insensitively
            if (char.ToLower(s[left]) != char.ToLower(s[right])) {
                return false;
            }
            
            left++;
            right--;
        }
        
        return true;
    }
}
